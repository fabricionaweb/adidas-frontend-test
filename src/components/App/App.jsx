import React from 'react';
import { Header } from '../Header';
import { Languages } from '../Languages';

export function App() {
  return (
    <main>
      <Languages />
      <Header lead="Sign up form" title="Get more information" />
    </main>
  );
}

export default App;
