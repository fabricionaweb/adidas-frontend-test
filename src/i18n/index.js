import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import resources from './resources';

i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources,
    lng: 'en', // default language
    fallbackLng: 'en', // use en if detected lng is not available
    keySeparator: false, // we do not use keys in form messages.welcome
    debug: process.env.NODE_ENV.includes('dev'),
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
  });

export default i18n;
