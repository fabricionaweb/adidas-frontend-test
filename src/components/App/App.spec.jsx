import React from 'react';
import { render } from '@testing-library/react';
import { App } from './App';
import '../../i18n';

describe('App.jsx', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
