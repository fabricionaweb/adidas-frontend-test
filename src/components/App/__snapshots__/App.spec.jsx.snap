// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.jsx should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="container"
    >
      <select
        data-testid="languages"
      >
        <option
          value="en"
        >
          English
        </option>
        <option
          value="de"
        >
          Germany
        </option>
      </select>
    </div>
  </div>
  <header
    class="wrapper"
  >
    <h1
      class="heading"
    >
      <p>
        Sign up form
      </p>
      Get more information
    </h1>
  </header>
  <section
    class="wrapper "
  >
    <form
      class="form"
    >
      <div
        class="row"
      >
        <label
          class="label"
          for="email"
        >
          Your e-mail 
          <input
            class="input"
            id="email"
            name="email"
            type="email"
          />
        </label>
      </div>
      <div
        class="row"
      >
        <div
          class="wrapper"
        >
          Select your gender
          <ul
            class="list"
          >
            <li>
              <label
                for="gender_0"
              >
                <input
                  class="input"
                  id="gender_0"
                  name="gender"
                  type="radio"
                  value="Male"
                />
                <span
                  class="label"
                >
                  Male
                </span>
              </label>
            </li>
            <li>
              <label
                for="gender_1"
              >
                <input
                  class="input"
                  id="gender_1"
                  name="gender"
                  type="radio"
                  value="Female"
                />
                <span
                  class="label"
                >
                  Female
                </span>
              </label>
            </li>
            <li>
              <label
                for="gender_2"
              >
                <input
                  class="input"
                  id="gender_2"
                  name="gender"
                  type="radio"
                  value="Non-binary"
                />
                <span
                  class="label"
                >
                  Non-binary
                </span>
              </label>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="row"
      >
        <label
          class="wrapper"
          for="confirmAge"
        >
          <input
            class="input"
            id="confirmAge"
            name="confirmAge"
            type="checkbox"
          />
          <span
            class="label"
          >
            Iâ€™m above a certain age 
          </span>
        </label>
      </div>
      <button
        class="button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
